sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/StringUtils.scala

```diff
@@ -81,16 +87,16 @@ object StringUtils extends Logging {
   /**
    * This utility can be used for filtering pattern in the "Like" of "Show Tables / Functions" DDL
    * @param names the names list to be filtered
-   * @param pattern the filter pattern, only '*' and '|' are allowed as wildcards, others will
-   *                follow regular expression convention, case insensitive match and white spaces
-   *                on both ends will be ignored
+   * @param pattern the filter pattern, only '*', '%', '_' and '|' are allowed as wildcards, others
+   *                will follow regular expression convention, case insensitive match and white
+   *                spaces on both ends will be ignored
    * @return the filtered names list in order
    */
   def filterPattern(names: Seq[String], pattern: String): Seq[String] = {
     val funcNames = scala.collection.mutable.SortedSet.empty[String]
     pattern.trim().split("\\|").foreach { subPattern =>
       try {
-        val regex = ("(?i)" + subPattern.replaceAll("\\*", ".*")).r
+        val regex = ("(?i)" + escapeLikeRegex(subPattern, '\\', true)).r
```

Comment:
`escapeLikeRegex(subPattern.replaceAll("\\*", "%", '\\', true)` ?

Reply:
> `escapeLikeRegex(subPattern.replaceAll("\\*", "%", '\\', true)` ? so we not need to change the behavior of `escapeLikeRegex`.

Goot suggestion, updated

Reply:
There are some conflict between these two logic...maybe we should follow other DB's rule.